src/main/java/org/opensearch/sdk/ExtensionRestPathRegistry.java

```diff
@@ -28,11 +28,11 @@ public class ExtensionRestPathRegistry {
      * Register a REST handler to handle a method and route in this extension's path registry.
      *
      * @param method  The method to register.
-     * @param uri  The URI to register. May include named wildcards.
+     * @param path  The path to register. May include named wildcards.
      * @param extensionRestHandler  The RestHandler to handle this route
      */
-    public void registerHandler(Method method, String uri, ExtensionRestHandler extensionRestHandler) {
-        String restPath = restPathToString(method, uri);
+    public void registerHandler(Method method, String path, ExtensionRestHandler extensionRestHandler) {
```

Comment:
I know this change has been done in the companion PR but can we keep it as `uri` instead of `path`?

Reply:
I did a lot of research before changing this.
 - This is a parallel of `RestController`'s registerHandler method which takes `path`.
https://github.com/opensearch-project/OpenSearch/blob/2bfe8b31af354e5d00e4ac67d544ce664f083934/server/src/main/java/org/opensearch/rest/RestController.java#L203

 - The `Route` class takes a `path` which is really what is being both registered _and_ handled
https://github.com/opensearch-project/OpenSearch/blob/3aef125d0dbf2f29d9bf00b84cb52892a27a5328/server/src/main/java/org/opensearch/rest/RestHandler.java#L185-L190

 - `uri` includes the query (e.g. the `?foo=bar&baz=qux` part) and that query part is not being registered
https://github.com/opensearch-project/OpenSearch/blob/4d3da00c1235fbceb65239336a027936ec781dc2/server/src/main/java/org/opensearch/http/HttpRequest.java#L65-L70

 - `path` and `uri` are the same if there is no query.  The query part gets parsed out to params:
https://github.com/opensearch-project/OpenSearch/blob/4d3da00c1235fbceb65239336a027936ec781dc2/server/src/main/java/org/opensearch/rest/RestRequest.java#L178-L180

So in theory if you *want* the URI including the query string in the Rest Handler, we could add it. But I think it duplicates the path + params combination.  More importantly , "path" is actually what is being registered and looked up here, and using the "uri" name is potentially confusing.